### DISCUSSION TITLE
How can I get the session data within axios interceptor without calling /api/auth/session ?

### DISCUSSION THREAD

**Discussion by fangwohung** [nextauthjs/next-auth#3294] [Help]
### Question 💬

I'm using next auth 4.0.0-beta.7. 
Even though I can get the session data within axios interceptor by calling getSession(), it will make an http request. 
If the client calls 3 APIs, it will execute 6 http requests due to getSession(). 
I just want to reduce the amount of calling APIs, is there any ways to avoid this ?

### How to reproduce ☕️

N/A

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
If you are out of the React lifecycles (eg cannot use the `SessionProvider`), you could cache the session and add deduplication to the outgoing requests. You are on your own though. This is out of scope currently.

**Reply by fangwohung** [reply]
Should I store the session data within window object ?
